alc/backends/coreaudio.cpp

```diff
@@ -552,6 +593,44 @@ bool CoreAudioPlayback::reset()
             err);
         return false;
     }
+    
+    if(!mDevice->Flags.test(ChannelsRequest))
+    {
+        UInt32 propSize{};
+        Boolean writable;
+        err = AudioUnitGetPropertyInfo(mAudioUnit,
+                                       kAudioUnitProperty_AudioChannelLayout,
+                                       kAudioUnitScope_Output,
+                                       OutputElement,
+                                       &propSize,
+                                       &writable);
+        AudioChannelLayout *layout = new AudioChannelLayout[propSize]();
+        
+        if(err == noErr)
+        {
+            err = AudioUnitGetProperty(mAudioUnit, kAudioDevicePropertyPreferredChannelLayout, kAudioUnitScope_Output, OutputElement, layout, &propSize);
+            
+            std::vector tagsArray = std::vector<AudioChannelLayoutTag>(layout->mNumberChannelDescriptions);
+            
+            for(UInt32 i=0; i<layout->mNumberChannelDescriptions; i++) {
+                tagsArray[i] = layout->mChannelDescriptions[i].mChannelLabel;
+            }
+            
+            if(err == noErr)
+            {
+                auto chaniter = std::find_if(chanmaps.cbegin(), chanmaps.cend(),
+                                             [&tagsArray](const ChannelMap &chanmap) -> bool
+                                             {
+                    return std::includes(tagsArray.begin(), tagsArray.end(), chanmap.map.begin(), chanmap.map.end());
```

Comment:
I don't think this works as intended. According to [cppreference](https://en.cppreference.com/w/cpp/algorithm/includes), the arrays need to be sorted:

> Returns `true` if the sorted range [`first2`, `last2`) is a [subsequence](https://en.wikipedia.org/wiki/subsequence) of the sorted range [`first1`, `last1`) (a subsequence need not be contiguous).
> 1) If [`first1`, `last1`) or [`first2`, `last2`) is not [sorted](https://en.cppreference.com/w/cpp/algorithm#Requirements) with respect to operator<(until C++20) [std::less](http://en.cppreference.com/w/cpp/utility/functional/less){}(since C++20), the behavior is undefined.

But I don't think the given enum values are guaranteed to be ordered.

Reply:
I've solved this by sorting the channel map that comes back from CoreAudio. (I don't believe Apple defines any particular order even though they seem to come back in ascending order.)

I also left a comment next to the constant values that any new values should be defined in ascending order. Probably a good tradeoff to prevent needing to sort those at runtime as well.